lib/cloudregister/registerutils.py

```diff
@@ -109,19 +109,23 @@ def check_registration(smt_server_name):
 # ----------------------------------------------------------------------------
 def clean_hosts_file(domain_name):
     """Remove the smt server entry from the /etc/hosts file"""
+    if isinstance(domain_name, str):
```

Comment:
Suggest using a pythonic try/except here. If this script runs on python2 and python3 you would need to handle different types (basestring for python2 and str for python3).

Reply:
We no longer support Python 2 with the registration client. So the concern with the multiple string types is somewhat academic. Never the less. The problem is that using `.encode()` generates a very generic "AttributeError" if domain_name is not a string. Thus, what would be caught in a "try - except" implementation a very generic error would be caught and that IMHO is not very intuitive in this case. Since `[ ].encode` produces the same as `b' '.encode()`.  I decided that showing the intent that domain_name should be a str, or could by bytes, would be more explicit in this case.

Reply:
Fair, my assumption if I see something like:

```
try:
    something = something.encode()
except AttributeError:
    pass
```

Is the same though. That the code is handling both bytes like objects and string like objects but in a duck typed way rather than the strict typed way.